### PR TITLE
Add generator hepmc3 products

### PR DESCRIPTION
**Pull request by mkirsano**
#### PR description:

Add generator products corresponding to HepMC3 event record. They are not yet added to edm

#### PR validation:

These products are not yet used.
Used in a local prototype of Pythia8Interface that uses both HepMC2 and HepMC3

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by smuzaffar** [comment]
code-checks with cms.week1.PR_0525f52f/52.0-78c28557adab64659c80be19da18577c

**Comment by mkirsano** [comment]
please test

File: SimDataFormats/GeneratorProducts/src/HepMC3Product.cc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+      TMatrixD p4lab(4, 1);
+      p4lab = tmplorentz * p4;
+      //std::cout << " momentum lorentz: " << p4lab(1,0) << " " << p4lab(3,0) << " " << p4lab(2,0) << std::endl;
+      part->set_momentum(HepMC3::FourVector(p4lab(1, 0), p4lab(3, 0), p4lab(2, 0), p4lab(0, 0)));
+    }
+
+    isPBoostApplied_ = true;
+  } else {
+    std::cout << " no type found for boostToLab(std::string), options are vertex or momentum" << std::endl;
```
Comment (qliphy) [inline SimDataFormats/GeneratorProducts/src/HepMC3Product.cc:104]: couts should be moved to LogPrint (cout is not thread safe)

**Comment by qliphy** [comment]
@mkirsano Are there any validation results available?

**Comment by iarspider** [comment]
please test with https://github.com/cms-sw/cmsdist/pull/7746

File: SimDataFormats/GeneratorProducts/src/HepMC3Product.cc (hunk 1 above)
Comment (civanch) [reply SimDataFormats/GeneratorProducts/src/HepMC3Product.cc:104]: @mkirsano , you may comment out cout for the time being and later add logVerbatim or logWarning if needed.

**Comment by mkirsano** [comment]
please test

**Comment by civanch** [comment]
@GurpreetSinghChahal, @SiewYan, can you, please, consider this PR comment or sign.

**Comment by qliphy** [comment]
ping @cms-sw/generators-l2

**Comment by Saptaparna** [comment]
+1

**Comment by qliphy** [comment]
+1
as discussed at yesterday's ORP

**Comment by qliphy** [comment]
We got an FWlite error
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/fwlite/slc7_amd64_gcc10/CMSSW_12_4_X_2022-04-20-1100/SimDataFormats/GeneratorProducts

@mkirsano please have a look. Thanks!

**Comment by smuzaffar** [comment]
@qliphy , this should be fixed by https://github.com/cms-sw/cmsdist/pull/7801 . As SimDataFormats/GeneratorProducts ( which is part of fwlite build) now depend on `hepmc3` so we should include it in fwlite distribution too.

**Pull request merged**